### ISSUE TITLE
attribute table - order by field type number

### ISSUE THREAD

**Issue by josemvm** [3liz/lizmap-web-client#1156]
hi

when i try to sort the table by gid, lizmap not parsing field number as number but as string, from postgreSQL tables

![imagem](https://user-images.githubusercontent.com/10053874/55283083-41e93a80-534a-11e9-8376-9a0d42520a14.png)

it's a regression...

lwc: all versions
qgis server 2.18 and 3.4

thanks

**Comment by josemvm** [comment]
please, can someone try with it:

```
CREATE TABLE sortable (
    gid integer NOT NULL,
    the_geom geometry(MultiPoint,3763),
    id integer
    );

INSERT INTO sortable(gid, the_geom, id)
VALUES 
	(1923, '0104000020B30E0000010000000101000000C687BC5E3C85CE40DE1173B0C667F9C0', 222),
	(1931, '0104000020B30E0000010000000101000000E6AB273EFA4ACE402590F0335829F9C0', 242),
	(1940, '0104000020B30E00000100000001010000007A44D0FB4A19CF404AACC5E1E72DF9C0', 2281),
	(2091, '0104000020B30E0000010000000101000000C37757A0ED3DD340488659846ECCF7C0', 3),
	(2244, '0104000020B30E0000010000000101000000145D7B19EB3DD340C45846A47ECCF7C0', 6),
	(7, '0104000020B30E0000010000000101000000CD9974B24A27CC409C25549BF3B2F7C0', 1),
	(132, '0104000020B30E0000010000000101000000B0A9343452A8CF40A32DCB6D2EABF7C0', 713);
```

thanks

**Comment by josemvm** [comment]
@rldhont it is ok on qgis server 2.14

**Comment by nboisteault** [comment]
I can confirm the issue with a 2.18 project with QGIS Server 3.4 but not QGIS Server 2.18.

**Comment by josemvm** [comment]
@nboisteault i'm using qgis server 2.18.28 on ubuntu 16.04.6 and it doesn't work

**Comment by nboisteault** [comment]
@josemvm Can you replace line 1300 in attributeTable.js by `if (idx in cTypes && (cTypes[idx] == 'integer' || cTypes[idx] == 'int'))` and tell me if it resolves the issue ?

**Comment by josemvm** [comment]
@nboisteault the problem remains on qgis server 2.18 and 3.4

**Comment by rldhont** [comment]
I have change the way QGIS 2.18 described numbers to have a better description of the field in the describe featuretype.

**Comment by rldhont** [comment]
@josemvm can you provide the describeFeatureType ?

**Comment by josemvm** [comment]
integer

**Comment by rldhont** [comment]
Can you provide the XML DesribeFeatureType of the layer ?

**Comment by josemvm** [comment]
sorry, how can i do this request?

**Comment by josemvm** [comment]
example:

```
</complexType>
<element name="Bocas_de_Lobo" type="qgs:Bocas_de_LoboType" substitutionGroup="gml:_Feature"/>
<complexType name="Bocas_de_LoboType">
<complexContent>
<extension base="gml:AbstractFeatureType"></extension>
</complexContent>
</complexType>
<element name="Colectores" type="qgs:ColectoresType" substitutionGroup="gml:_Feature"/>
<complexType name="ColectoresType">
<complexContent>
<extension base="gml:AbstractFeatureType">
<sequence>
<element minOccurs="0" maxOccurs="1" name="geometry" type="gml:GeometryPropertyType"/>
<element alias="GID" name="gid" type="int"/>
<element alias="Tipo" nillable="true" name="a0101" type="string"/>
<element alias="Material" nillable="true" name="a0103" type="string"/>
<element alias="Material Obs." nillable="true" name="a0103_obs" type="string"/>
<element alias="Escoamento" nillable="true" name="a0105" type="string"/>
<element alias="Juntas" nillable="true" name="a0107" type="string"/>
<element alias="Secção" nillable="true" name="seccao" type="string"/>
<element alias="Ø (mm)" nillable="true" name="a0104" type="string"/>
<element alias="Câm./montante" nillable="true" name="idmon" type="int"/>
<element alias="Saídas Col.M" nillable="true" name="idmoncol" type="string"/>
<element alias="Câm./jusante" nillable="true" name="idjus" type="int"/>
<element alias="Saídas Col.J" nillable="true" name="idjuscol" type="string"/>
<element alias="Traçado" nillable="true" name="tracado" type="string"/>
<element alias="Observações" nillable="true" name="obs" type="string"/>
<element alias="Compr. (m)" nillable="true" name="comp" type="decimal"/>
</sequence>
</extension>
</complexContent>
</complexType>
```

**Comment by rldhont** [comment]
Thanks! Number types are int and decimal.

**Comment by nboisteault** [comment]
I'v updated QGIS server from 2.18.24 to 2.18.28 and I have now 'int' and not 'integer'. You were right @josemvm

**Comment by nboisteault** [comment]
Fixed with #1165
